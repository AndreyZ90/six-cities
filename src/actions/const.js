const GET_OFFERS = `GET_OFFERS`;
const CHANGE_CURRENT_CITY = `CHANGE_CURRENT_CITY`;

export {
  GET_OFFERS,
  CHANGE_CURRENT_CITY
};
