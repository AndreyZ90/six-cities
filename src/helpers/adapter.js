export default class Adapter {
  static offer(offer) {
    return {
      id: offer.id,
      isFavorite: offer.is_favorite,
      isPremium: offer.is_premium,
      previewImage: offer.preview_image,
      price: offer.price,
      rating: offer.rating,
      title: offer.title,
      type: offer.type,
      images: offer.images,
      bedrooms: offer.bedrooms,
      maxAdults: offer.max_adults,
      goods: offer.goods,
      host: {
        avatarUrl: offer.host.avatar_url,
        id: offer.host.id,
        isPro: offer.host.is_pro,
        name: offer.host.name
      },
      description: offer.description,
      location: offer.location,
      city: offer.city
    };
  }

  static offers(offers) {
    return offers.map((data) => this.offer(data));
  }

  static review(review) {
    return {
      comment: review.comment,
      date: review.date,
      id: review.id,
      rating: review.rating,
      user: {
        avatarUrl: review.user.avatar_url,
        id: review.user.id,
        isPro: review.user.is_pro,
        name: review.user.name
      }
    };
  }

  static reviews(reviews) {
    return reviews.map((data) => this.review(data));
  }
}
