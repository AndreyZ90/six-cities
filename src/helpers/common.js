const convertRatingToStyle = (rating) => {
  const maxStars = 5;
  const maxPercent = 100;
  const step = maxPercent / maxStars;

  return rating < 1 ? step : Math.round(rating) * step;
};

const getGeoCoords = (data) => {
  if (data instanceof Array) {
    return data.map(({location}) => [location.latitude, location.longitude]);
  }

  if (!(data instanceof Array) && data instanceof Object) {
    if (!data) {
      return [];
    }

    return [data.location.latitude, data.location.longitude];
  }

  return [];
};

export {convertRatingToStyle, getGeoCoords};
