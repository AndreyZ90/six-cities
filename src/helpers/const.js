const ApiConfig = {
  baseURL: `https://4.react.pages.academy/six-cities`,
  timeout: 5000
};

const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const ClassPrefix = {
  CARD: `place-card`,
  CITY: `cities`,
  DETAILS: `property`,
  REVIEWS: `reviews`,
  NEARBY: `near-places`
};

const HouseType = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

const MapPin = {
  src: `/img/pin.svg`,
  size: [30, 40]
};

const MapPinActive = {
  src: `/img/pin-active.svg`,
  size: [30, 40]
};

const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`
};

export {
  ApiConfig,
  AuthStatus,
  ClassPrefix,
  HouseType,
  MapPin,
  MapPinActive,
  SortType
};
