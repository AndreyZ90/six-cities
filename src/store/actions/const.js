const ActionType = {
  FETCH_DATA_REQUEST: `FETCH_DATA_REQUEST`,
  FETCH_DATA_SUCCESS: `FETCH_DATA_SUCCESS`,
  FETCH_OFFERS_SUCCESS: `FETCH_OFFERS_SUCCESS`,
  SET_CITIES: `SET_CITIES`,
  SET_CURRENT_CITY: `SET_CURRENT_CITY`,
  SET_CURRENT_SORT: `SET_CURRENT_SORT`,
  FETCH_LOGIN_SUCCESS: `FETCH_LOGIN_SUCCESS`,
  FETCH_LOGIN_FAILURE: `FETCH_LOGIN_FAILURE`,
  FETCH_OFFERS_NEARBY_SUCCESS: `FETCH_OFFERS_NEARBY_SUCCESS`,
  FETCH_REVIEWS_SUCCESS: `FETCH_REVIEWS_SUCCESS`
};

export default ActionType;
