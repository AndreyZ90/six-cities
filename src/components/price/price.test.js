import React from 'react';
import TestRenderer from 'react-test-renderer';

import Price from '@/components/price/price';

import {ClassPrefix} from '@/helpers/const';

const price = 300;

describe(`Price component snapshot`, () => {
  test(`Should correctly render Price component (default)`, () => {
    const tree = TestRenderer.create(<Price price={price} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render Price component (with prefix CARD)`, () => {
    const tree = TestRenderer.create(<Price price={price} classPrefix={ClassPrefix.CARD} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`Should correctly render Price component (with prefix DETAILS)`, () => {
    const tree = TestRenderer.create(<Price price={price} classPrefix={ClassPrefix.DETAILS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
