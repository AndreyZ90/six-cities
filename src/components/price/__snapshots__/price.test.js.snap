// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price component snapshot Should correctly render Price component (default) 1`] = `
<div
  className="place-card__price"
>
  <b
    className="place-card__price-value"
  >
    €
    300
  </b>
  <span
    className="place-card__price-text"
  >
     / night
  </span>
</div>
`;

exports[`Price component snapshot Should correctly render Price component (with prefix CARD) 1`] = `
<div
  className="place-card__price"
>
  <b
    className="place-card__price-value"
  >
    €
    300
  </b>
  <span
    className="place-card__price-text"
  >
     / night
  </span>
</div>
`;

exports[`Price component snapshot Should correctly render Price component (with prefix DETAILS) 1`] = `
<div
  className="property__price"
>
  <b
    className="property__price-value"
  >
    €
    300
  </b>
  <span
    className="property__price-text"
  >
      night
  </span>
</div>
`;
